src/rdf_validator/core.clj

```diff
@@ -113,24 +118,33 @@
     :parse-fn parse-variables-file
     :default {}]])
 
-(defn -main
-  [& args]
+(defn inner-main 
+  [args]
   (let [{:keys [errors options] :as result} (cli/parse-opts args cli-options)]
     (if (nil? errors)
-      (try
-        (let [suite-sources (concat (:suite options) (tc/classpath-test-suite-sources))
-              endpoint (create-endpoint options)
-              query-variables (:variables options)
-              suites (tc/resolve-test-suites suite-sources)
-              suites-to-run (:arguments result)
-              test-cases (tc/suite-tests suites suites-to-run)
-              test-reporter (reporting/->ConsoleTestReporter)
-              {:keys [failed errored] :as test-summary} (run-test-cases test-cases query-variables endpoint test-reporter)]
-          (System/exit (+ failed errored)))
-        (catch Exception ex
-          (binding [*out* *err*]
-            (println (.getMessage ex))
-            (.printStackTrace ex)
-            (System/exit 1))))
-      (do (invalid-args result)
-          (System/exit 1)))))
+      (let [suite-sources (concat (:suite options) (tc/classpath-test-suite-sources))
+            endpoint (create-endpoint options)
+            query-variables (:variables options)
+            suites (tc/resolve-test-suites suite-sources)
+            suites-to-run (:arguments result)
+            test-cases (tc/suite-tests suites suites-to-run)
+            test-reporter (reporting/->ConsoleTestReporter)]
+        (run-test-cases test-cases query-variables endpoint test-reporter))
+      (throw (ex-info "Invalid command line arguments" {:type :invalid-cli-arguments
+                                                        :cli-result result})))))
+
+(defn -main
+  [& args]
+  (try
+    (set-missing-value-formatter! on-missing-query-template-variable)
```

Comment:
Should this be in `inner-main`? We might want to add options around this behaviour if different clients want to configure it.

Reply:
Should we wait until we reconfigure the rdf-validator to move it there?

Reply:
ok!